egs/wsj/s5/steps/nnet3/chain/train_more.py

```diff
@@ -0,0 +1,590 @@
+    # note: the '+ args.frame_subsampling_factor / 2' is to allow for the
+    # fact that we'll be shifting the data slightly during training to give
+    # variety to the training data.
+    egs_left_context_initial = (left_context_initial + args.frame_subsampling_factor / 2 if
+                                left_context_initial >= 0 else -1)
+    egs_right_context_final = (right_context_final + args.frame_subsampling_factor / 2 if
+                               right_context_final >= 0 else -1)
+
+    default_egs_dir = '{0}/egs'.format(args.dir)
+    if (args.stage <= -3) and args.egs_dir is None:
+        logger.info("Generating egs")
+        # this is where get_egs.sh is called.
+        chain_lib.generate_chain_egs(
+            dir=args.dir, data=args.feat_dir,
+            lat_dir=args.lat_dir, egs_dir=default_egs_dir,
+            left_context=egs_left_context,
+            right_context=egs_right_context,
+            left_context_initial=egs_left_context_initial,
+            right_context_final=egs_right_context_final,
+            run_opts=run_opts,
+            left_tolerance=args.left_tolerance,
+            right_tolerance=args.right_tolerance,
+            frame_subsampling_factor=args.frame_subsampling_factor,
+            alignment_subsampling_factor=args.alignment_subsampling_factor,
+            frames_per_eg_str=args.chunk_width,
+            srand=args.srand,
+            egs_opts=args.egs_opts,
+            cmvn_opts=args.cmvn_opts,
+            online_ivector_dir=args.online_ivector_dir,
+            frames_per_iter=args.frames_per_iter,
+            transform_dir=args.transform_dir,
+            stage=args.egs_stage)
+
+    if args.egs_dir is None:
+        egs_dir = default_egs_dir
+    else:
+        egs_dir = args.egs_dir
+
+    [egs_left_context, egs_right_context,
+     frames_per_eg_str, num_archives] = (
+        common_train_lib.verify_egs_dir(egs_dir, feat_dim,
+                                        ivector_dim, ivector_id,
+                                        egs_left_context, egs_right_context,
+                                        egs_left_context_initial,
+                                        egs_right_context_final))
+    assert(os.path.getctime('{0}/cegs.1.ark'.format(egs_dir)) >
+           os.path.getctime('{0}/phone_lm.fst'.format(args.dir)))
+    assert(args.chunk_width == frames_per_eg_str)
+    num_archives_expanded = num_archives * args.frame_subsampling_factor
+
+    if (args.num_jobs_final > num_archives_expanded):
+        raise Exception('num_jobs_final cannot exceed the '
+                        'expanded number of archives')
+
+    # copy the properties of the egs to dir for
+    # use during decoding
+    logger.info("Copying the properties from {0} to {1}".format(egs_dir, args.dir))
+    common_train_lib.copy_egs_properties_to_exp_dir(egs_dir, args.dir)
+
+    if (args.stage <= -2) and os.path.exists(args.dir+"/configs/init.config"):
```

Comment:
This stage is not required in this script. But you should check that init.raw exists.